jl4-lsp/src/LSP/L4/Viz/Ladder.hs

```diff
@@ -133,12 +163,22 @@ translateExpr False = go
           V.Or uid <$> traverse go (scanOr e)
         Where _ e' _ds -> go e' -- TODO: lossy
 
+        -- TODO: Should handle BoolLits differently too
         -- 'var'
         App _ resolved [] ->
           leafFromVizName (mkVizNameWith prettyLayout resolved)
-        -- TODO: Will be replacing this temporary version with a variant for App on the frontend
-        App _ (getOriginal -> MkName _ fnName) args ->
-          leaf "" $ Text.unwords (rawNameToText fnName : (prettyLayout <$> args))
+
+        fnApp@(App _ fnResolved@(getOriginal -> fnName) args) -> do
+          fnOfAppisFnFromBooleanToBooleans <- and <$> traverse exprHasBooleanType (fnApp : args)
```

Comment:
I think it would've been better to pass in the annos directly here.

Reply:
This does help make it more readable, thank you!